sql/parser: make NormalizableTableReference persist results of formatting

knz opened the pull request:
This is the prefix commit to  #17310, split off to facilitate reviewing.

Prior to this patch, if a NormalizableTableReference was normalized
as part of the work done by FmtReformatTables, the result of normalization
would be discarded because the Format() method took a copy.

This change makes Format() take a reference, so that the result
of format-based normalization stay in the AST.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17315)
<!-- Reviewable:end -->

justinj commented on pkg/sql/parser/table_name.go:
```diff
@@ -261,7 +261,7 @@ type TableNameWithIndex struct {
 
 // Format implements the NodeFormatter interface.
 func (n TableNameWithIndex) Format(buf *bytes.Buffer, f FmtFlags) {
```
Should this (and anything else that contains a `NormalizableTableName`) also be a reference? I'm not sure when this can come up but could we also want this to persist its normalization?

knz replied:
oh you're absolutely right. Well spotted. Will fix.

knz replied:
Done, PTAL. The changes in  #17310 will exercise this.

justinj reviewed (commented):
LGTM modulo question

knz committed:
sql/parser: make NormalizableTableReference persist results of formatting

Prior to this patch, if a NormalizableTableReference was normalized
as part of the work done by FmtReformatTables, the result of normalization
would be discarded because the Format() method took a copy.

This change makes Format() take a reference, so that the result
of format-based normalization stay in the AST.

justinj commented:
LGTM

---

Review status: 0 of 11 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17315#-:-KqT4n6L24Caa4Ejp8vf:bb6ywmx)*
<!-- Sent from Reviewable.io -->

knz commented:
TFYR!
